vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+
+These functions are available in `admiral`, below examples show its usage for `ADTR` 
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      order = vars(ADY),
+      new_var = ABLFL,
+      mode = "last"
+    ),
+    filter = ADY <= 1
+  ) %>%
+  derive_var_base(
+    by_vars = vars(USUBJID)
+  )
+```
+
+## Derive `NADIR` {#nadir}
+
+Once we have a TSD parameter, the NADIR is set to the lowest sum of diameters in
+Analysis Value before the current observation, where Analysis Flag 01 =Y.  The
+first observations after baseline (depending on how you have defined baseline
+earlier) is set to analysis value where baseline record flag = Y
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_vars_joined(
+    dataset_add = adtr_sum,
+    by_vars = vars(USUBJID),
+    order = vars(AVAL),
+    new_vars = vars(NADIR = AVAL),
+    join_vars = vars(ADY),
+    filter_add = ADY > 0 & ANL01FL == "Y",
+    filter_join = ADY.join < ADY,
+    mode = "first",
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, AVISIT, AVAL, NADIR)
+)
+```  
+
+## Derive Change from Baseline (`CHG`, `PCHG`) {#chg}
+
+These functions are available in admiral, below examples show its usage for
+`ADTR`.  We calculate the change from baseline and percentage change from
+baseline for the parameter TSD and the variable `NADIR`.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_chg() %>%
+  derive_var_pchg() %>%
+  mutate(
+    CHGNAD = AVAL - NADIR,
+    PCHGNAD = 100 * CHGNAD / NADIR
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHG, CHGNAD, PCHGNAD)
+)
+```  
+
+## Derive Additional Flag Variables {#addflag}
+
+Depending on the analysis flags you require, you may want to create additional
+flag variables that will then aid setting up any additional analysis flag
+variables defined as per your company specifications.
+
+###  Derive `CRFL`, `CRNFL`
+
+In this example, `CRFL` flags where the patient has had Complete Response, and
+`CRNFL`, flags where Complete response contributed to the NADIR calculation.
+
+```{r}
+adtr_sum <- adtr_sum %>% mutate(
+  CRFL = if_else(AVAL == 0 & ANL01FL == "Y", "Y", NA_character_),
+  CRNFL = if_else(NADIR == 0, "Y", NA_character_)
+)
+```
+
+### Derive `PDFL`
+
+In this example, we want to flag when a patient has had PD.  This could be done
+in a number of ways:
+
+1. take the date of PD from `ADRS`
+
+2. take the first date when the overall response is PD from `RS` or
+
+3. Calculate from source data, the derivation for PD for target lesions is based
+on the calculated sum of diameters of the assessed lesions results in a
+calculated Percent Change from NADIR >=20% and a calculated Change from NADIR
+>=5mm depending on your company specifications.
```

Comment:
Should we add the complete response criterion (no CR at the current observation and CR at the nadir observation indicates PD)?

Reply:
I think this for data checking purposes for PD to cehck if nadir>=20% etc. do we really need to do here? I would suggest somewhat unconventional to add investigator( or independent reviewer) responses from RS in ADTR for each visit.

Reply:
we have scenarios where we need to program from scratch.
ours specs have optional RS or ADRS

Reply:
@bundfussr Stefan ,  I am okay to resolve this comment, if its okay for you can resolve